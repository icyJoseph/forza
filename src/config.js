// endpoints
export const url =
  process.env.NODE_ENV === "production"
    ? process.env.REACT_APP_SECRET
    : "http://localhost:1337/";
