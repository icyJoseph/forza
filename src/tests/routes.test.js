import React from "react";
import { shallow } from "enzyme";
import Routes, {
  TopMenu,
  League,
  Landing,
  FloatingActionButtons
} from "../routes";
import { url } from "../config";

describe("static copy of the routes", () => {
  it("matches snapshot", () => {
    expect(shallow(<Routes />)).toMatchSnapshot();
  });
});

describe("TopMenu", () => {
  const loaded = shallow(<TopMenu />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("League", () => {
  const loaded = shallow(<League />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("Landing", () => {
  const loaded = shallow(<Landing />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("FloatingActionButtons", () => {
  const loaded = shallow(<FloatingActionButtons />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("builds the correct endpoint for development", () => {
  expect(url).toEqual("http://localhost:1337/");
});
