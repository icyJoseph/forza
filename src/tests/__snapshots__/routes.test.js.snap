// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`static copy of the routes matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Router />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown>
        <Route
                component={[Function]}
                path="/:league?"
        />
        <Switch>
                <Route
                        component={[Function]}
                        exact={true}
                        path="/:league"
                />
                <Route
                        component={[Function]}
                        path="/"
                />
        </Switch>
        <Route
                component={[Function]}
                path="/:league"
        />
</Unknown>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/:league?"
/>,
          <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/:league"
            />
            <Route
                        component={[Function]}
                        path="/"
            />
</Switch>,
          <Route
            component={[Function]}
            path="/:league"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/:league?",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/:league"
/>,
              <Route
                component={[Function]}
                path="/"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/:league",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/:league",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown>
          <Route
                    component={[Function]}
                    path="/:league?"
          />
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/:league"
                    />
                    <Route
                              component={[Function]}
                              path="/"
                    />
          </Switch>
          <Route
                    component={[Function]}
                    path="/:league"
          />
</Unknown>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/:league?"
/>,
            <Switch>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/:league"
              />
              <Route
                            component={[Function]}
                            path="/"
              />
</Switch>,
            <Route
              component={[Function]}
              path="/:league"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/:league?",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/:league"
/>,
                <Route
                  component={[Function]}
                  path="/"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/:league",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/:league",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
